packages/apify/src/charging.ts

```diff
@@ -117,12 +135,48 @@ export class ChargingManager {
         return this.pricingModel === 'PAY_PER_EVENT';
     }
 
+    private loadPricingInfo(
+        pricingInfo: ActorRunPricingInfo | undefined,
+        maxTotalChargeUsd: number | undefined,
+    ) {
+        this.pricingModel = pricingInfo?.pricingModel;
+
+        // Load per-event pricing information
+        if (pricingInfo?.pricingModel === 'PAY_PER_EVENT') {
+            for (const [eventName, eventPricing] of Object.entries(
+                pricingInfo.pricingPerEvent.actorChargeEvents,
+            )) {
+                this.pricingInfo[eventName] = {
+                    price: eventPricing.eventPriceUsd,
+                    title: eventPricing.eventTitle,
+                };
+            }
+
+            this.maxTotalChargeUsd =
+                maxTotalChargeUsd ?? this.maxTotalChargeUsd;
+        }
+    }
+
+    private loadChargedEventCounts(
+        chargedEventCounts: Record<string, number> | undefined,
+    ) {
+        this.chargingState = {};
+
+        for (const [eventName, chargeCount] of Object.entries(
+            chargedEventCounts ?? {},
+        )) {
+            this.chargingState[eventName] = {
+                chargeCount,
+                totalChargedAmount:
+                    chargeCount * (this.pricingInfo[eventName]?.price ?? 0),
+            };
+        }
+    }
+
     /**
      * Initialize the ChargingManager by loading pricing information and charging state via Apify API.
      */
     async init(): Promise<void> {
-        this.chargingState = {};
-
         // Retrieve pricing information
         if (this.isAtHome) {
```

Comment:
We should be able to get rid of this with a future refactor so we could inject the pricing locally